maker-bundle/src/Maker/AbstractFragmentMaker.php

```diff
@@ -0,0 +1,174 @@
+{
+    protected ContaoFramework $framework;
+    protected TemplateGenerator $templateGenerator;
+    protected ClassGenerator $classGenerator;
+    protected DcaGenerator $dcaGenerator;
+    protected LanguageFileGenerator $languageFileGenerator;
+    protected ContaoDirectoryLocator $contaoDirectoryLocator;
+
+    public function __construct(ContaoFramework $framework, TemplateGenerator $templateGenerator, ClassGenerator $classGenerator, DcaGenerator $dcaGenerator, LanguageFileGenerator $languageFileGenerator, ContaoDirectoryLocator $contaoDirectoryLocator)
+    {
+        $this->framework = $framework;
+        $this->templateGenerator = $templateGenerator;
+        $this->classGenerator = $classGenerator;
+        $this->dcaGenerator = $dcaGenerator;
+        $this->languageFileGenerator = $languageFileGenerator;
+        $this->contaoDirectoryLocator = $contaoDirectoryLocator;
+    }
+
+    public function interact(InputInterface $input, ConsoleStyle $io, Command $command): void
+    {
+        $requiredValidator = static function ($input) {
+            if (null === $input || '' === $input) {
+                throw new RuntimeCommandException('This value cannot be blank');
+            }
+
+            return $input;
+        };
+
+        $definition = $command->getDefinition();
+
+        // Ask whether an empty dca palette should be created
+        $command->addArgument('addEmptyDcaPalette', InputArgument::OPTIONAL);
+        $question = new ConfirmationQuestion('Do you want to add an empty DCA palette?', true);
+        $input->setArgument('addEmptyDcaPalette', $io->askQuestion($question));
+
+        $command->addArgument('category', InputArgument::OPTIONAL, 'Choose a category');
+        $argument = $definition->getArgument('category');
+
+        $categories = $this->getExistingCategories();
+
+        $io->writeln(' <fg=green>Suggested Categories:</>');
+        $io->listing($categories);
+
+        $question = new Question($argument->getDescription());
+        $question->setValidator($requiredValidator);
+        $question->setAutocompleterValues($categories);
+
+        $input->setArgument('category', $io->askQuestion($question));
+
+        // Ask whether language files should be generated
+        $command->addArgument('addTranslation', InputArgument::OPTIONAL);
+        $question = new ConfirmationQuestion('Do you want to add translations?', true);
+        $input->setArgument('addTranslation', $io->askQuestion($question));
+
+        if ($input->getArgument('addTranslation')) {
+            $command
+                ->addArgument('translatedName', InputArgument::OPTIONAL, 'Choose an english name for this element')
+                ->addArgument('translatedDescription', InputArgument::OPTIONAL, 'Choose an english description for this element')
+                // FIXME: ->addArgument('languages', InputArgument::OPTIONAL, 'What languages do you want the element name translate to? (e.g. <fg=yellow>en, fr</>)')
```

Comment:
This has been commented, so I have added a `FIXME` mark. Is this the reason why the PR is still a draft?

Reply:
May because of `Things to consider`?